# Outras ferramentas de avaliação de investimentos

##Índice de Lucratividade (IL)

O Índice de lucratividade avalia a capacidade do investimento/projeto em gerar lucros. Em projetos que apresentam VLP's idênticos ou próximos, o IL poderá ser uma forma de identificar aquele que gerará maiores resultados econômicos. 

**Expressão Índice de Lucratividade**

![IL](https://lh4.googleusercontent.com/4MpJDkKBjPHVx55sz04-JK7gkf6h-XNTjCNgV7bKztuRgdBVnqpLyzPHyBcvOTJlYFRXMqQ9EZgHw5iAXFCy-OKa6O2Kd4VdkzPTz90YiD6hErAMcdrPbDgDWXqdUHHgEfBPO0Gs)

In [ ]:
import numpy as np

In [ ]:
cf1 = [-300, 140 , 160, 200, 43.50]
cf2 = [-600, 220 , 150, 615.45, 43.50]
rate = 0.2
nper = np.linspace(0, 4, 5, dtype=int)

In [ ]:
def vpl(cf, rate, nper):
  '''
  vp: lista com investimento inicial, onde será adicionado os valores presentes do fluxo de caixa.
  cf: valor do fluxo de caixa.
  rate: taxa de capitalização
  nper: período de capitalizaçãão
  '''
  vp = []
  for i in nper:
    if i == 0:
      vp.append(cf[i])
    elif i != 0:
      npv = round(cf[i] / np.power((1+rate),i),2)
      vp.append(npv)

  return sum(vp)

In [ ]:
#vpl cashflow 1
round(vpl(cf1, rate, nper),1)

64.5

In [ ]:
#vpl cashflow 2
round(vpl(cf2, rate, nper),1)

64.6

In [ ]:
def ind_lucr(cf, rate, nper):
  '''
  vp: lista com investimento inicial, onde será adicionado os valores presentes do fluxo de caixa.
  cf: valor do fluxo de caixa.
  rate: taxa de capitalização
  nper: período de capitalizaçãão
  '''
  vp = []
  for i in nper:
    if i != 0:
      npv = round(cf[i] / np.power((1+rate),i),2)
      vp.append(npv)
  
  il = sum(vp)/cf[0]

  return il * -1

In [ ]:
#indice de lucratividade cashflow 1

round(ind_lucr(cf1, rate, nper),4)

1.215

In [ ]:
#indice de lucratividade cashflow 2

round(ind_lucr(cf2, rate, nper), 4)

1.1077

Logo, podemos identificar que o projeto do cashflow 1 é mais lucrativo que o projeto do cashflow 2. 

## Custo Anual Equivalente

O custo anual equivalente ou CAE, permite conhecer os custos anuais atrelados ao investimento inicial. A ideia principal é que os investimentos sejam rateados uniformemente entre custos de investimentos, operação e oportunidade. 

**Expressão CAE**

![cae](http://www.portalaction.com.br/files/tex/333f9c2f8316333913df95571a0998950552b040.png)

**Exemplo 2**: 

![exemplo CAE](https://raw.githubusercontent.com/levisouuza/engenharia-economica-python/master/images/exemplo%20CAE.png?token=AO4A5ZTO3M5K6MF4XDCBUB267CTC4)

In [ ]:
CFa = [500, 120 , 120, 120]
CFb = [600, 100 , 100, 100, 100]

nperA = np.linspace(0, 3, 4, dtype=int)
nperB = np.linspace(0, 4, 5, dtype=int)

rate = 0.1

In [ ]:
vplA = vpl(CFa, rate, nperA)
vplA

798.42

In [ ]:
vplB = round(vpl(CFb, rate, nperB),2)
vplB

916.98

Apresentando os valores do VPL dos custos de A e B, a aquisição tende a ser a do projeto A. Entretanto, deve ser avaliado também as vidas úteis dos projetos em análise. Nesse caso, iremos utilizar a fórmula do CAE para verificar os custos dos projetoos evidenciando suas respectivas vidas úteis.

In [ ]:
def CAE(vp, rate, nper):
  cae = vp * ((((1+rate)**nper) * rate) / (((1+rate)**nper)-1))
  return round(cae, 2)

In [ ]:
CAEa = CAE(vplA, rate, 3)

CAEa

321.06

In [ ]:
CAEb = CAE(vplB, rate, 4)

CAEb

289.28

Comparando CAEa X CAEb, é possível verificar que o projeto B apresenta menor custo anual. Portanto, deve ser o aprovado. 

##Índice de Benefício/Custo

É relação entre o fluxo esperado de benefícios e o fluxo esperado de custos de investimento, ou seja, é quanto há de ganho de capital sobre o valor do investimento. 

**Expressão Índice de Benefício/Custo**

![IBC](https://cdn.shortpixel.ai/client/q_lossless,ret_img,w_699/https://negociovencedor.com/wp-content/uploads/2020/01/formula-indice-beneficio-custo-ibc-financa.jpg)



*   IBC > 1, Lucro Econômico
*   IBC < 1, Prejuízo Econômico



**Exemplo 3**: O engenheiro-chefe de uma fábrica de parafusos considera adquirir um novo torno digital que aumentará a produção durante os seus 5 anos de vida útil. A máquina custará 50.000,00 e acrescerá anualmente ao fluxo de caixa o valor de R$20.000,00. A diretoria só irá aprovar a aquisição se tiver 20% de retorno sobre o investimento. Será que é viável a compra desta nova máquina?

In [ ]:
CF = [50000, 20000, 20000, 20000, 20000, 20000]
rate = 0.2
nper = np.linspace(0, 5, 6, dtype=int)

In [ ]:
FluxoInvest = CF[0]

In [ ]:
FluxoBenef = round(vpl(CF, rate, nper)- CF[0],2)

FluxoBenef

59812.24

In [ ]:
ibc = round(FluxoBenef / FluxoInvest,4)

ibc

1.1962

O valor de 1.1962 significa que para cada R$ 1,00 investido, o investimento irá ter 0.1962 de lucro recebido. 

## Múltiplos


Para entender o conceito de múltiplos é necessário conhecer a definição de análise fundamentalista. 

Análise fundamentalista é o tipo de análise de investimento que utiliza fundamentos das empresas para sua avaliação. Os fundamentos são indicadores financeiros provenientes de sua operação e mercado de ações. 

Os múltiplos são os indicadores utilizados para esse tipo de análise. São frequentemente calculados para avaliações de empresas com o intuito de compra/venda de ações. 

Os mais utilizados são: 

*  PL
*  ROE
*  DY

#### 1.PL 

O PL ou preço/lucro é o indicador mais conhecido do Mercado Financeiro. O **P** é referente ao preço atual da ação que pode ser visto no home broker de sua corretora. O **L** é referente ao lucro dos últimos 12 meses por ação. 

O PL indica quanto tempo você terá o valor investido de volta. Quanto menor, melhor. Todavia, *retornos do passado não são garantias de retornos do futuro*, portanto, o PL **indica** que poderá ter um retorno em determinado período, mas não o garante.

#### 2.ROE 

ROE significa *RETURN ON EQUITY* ou Retorno sobre o Patrimônio Líquido. O ROE representa a rentabilidade adquirida pela empresa/investimento. É definido pela expressão:

ROE = Lucro Líquido(Últ.12 meses) / Patrimônio Líquido

Como regra de bolso, considere uma boa empresa com ROE acima de 10% ~ 15%.

#### 3.DY 

O DY ou Dividend Yield é o múltiplo fundamentalista que informa o quanto a empresa pagou de proventos(dividendos ou jsp) nos últimos 12 meses em relação ao seu *price* atual.

Os proventos são parcelas dos lucros das empresas em que os acionistas tem o direito de receber. Portanto, qual maior o DY, maiores são retornos dos acionistas.

No mercado acionário existe a possibilidade de ganhar dinheiro pela valorição do ativo ou pelos proventos pagos periodicamente pelas empresas. Portanto, o DY é um múltiplo muito importante para aqueles que desejam obter esse tipo de rendimento.


